### DISCUSSION TITLE
Add component to entity using world possible?

### DISCUSSION THREAD

**Discussion by Schr3da** [bevyengine/bevy#6758] [Q&A] [answered]
Hey,

sorry but is there a way to add/insert a component without using Commands?
Basically what I would like to do is to add an component to an existing entity by using the world

`use bevy_ecs::prelude::*;

use crate::prelude::*;

pub fn set_display_cells(rows: i32, columns: i32, world: &mut World) {
    let mut query = world.query::<(Entity, With<DisplayCell>)>();

    for (entity, _) in query.iter(world) {
        // how to add component to entity?

        // world.insert(entity, component) - does not existV
        // world.entity(entity).insert(SomeComponent) - does not work
    }

    for row in 0..rows {
        for column in 0..columns {
            world
                .spawn(DisplayCell {})
                .insert(Position { x: row, y: column });
        }
    }
}

` 

any help would be great.

regards
Schr3da

**Comment by james7132** [comment, marked as answer]
If you know an entity already exists, you can use [`World::entity_mut`](https://docs.rs/bevy_ecs/0.9.0/bevy_ecs/world/struct.World.html#method.entity_mut) to get an `EntityMut` for that entity.


```rust
world
    .entity_mut(entity)
    .insert(Position { x: row, y: column });
```

**Reply by Schr3da** [reply]
in bevy_ecs 0.9.0 it says it is unsafe + expected 0 args. are you refering to another becy_ecs version?

**Reply by hymm** [reply]
you're looking at `entities_mut`, not `entity_mut`

**Reply by Schr3da** [reply]
Thanks worked
